catalyst/contrib/dl/callbacks/kornia_transform.py

```diff
@@ -0,0 +1,217 @@
+                return self.model(
+                    batch[0].to(self.device).view(batch[0].size(0), -1)
+                )
+
+            def _handle_batch(self, batch):
+                # model train/valid step
+                x, y = batch
+                y_hat = self.model(x.view(x.size(0), -1))
+
+                loss = F.cross_entropy(y_hat, y)
+                accuracy01, *_ = metrics.accuracy(y_hat, y)
+                self.batch_metrics.update(
+                    {"loss": loss, "accuracy01": accuracy01}
+                )
+
+                if self.is_train_loader:
+                    loss.backward()
+                    self.optimizer.step()
+                    self.optimizer.zero_grad()
+
+        model = torch.nn.Linear(28 * 28, 10)
+        optimizer = torch.optim.Adam(model.parameters(), lr=0.02)
+
+        loaders = {
+            "train": DataLoader(
+                MNIST(os.getcwd(), train=True, transform=ToTensor()),
+                batch_size=32,
+            ),
+            "valid": DataLoader(
+                MNIST(os.getcwd(), train=False, transform=ToTensor()),
+                batch_size=32,
+            ),
+        }
+        transrorms = [
+            augmentation.RandomAffine(degrees=(-15, 20), scale=(0.75, 1.25)),
+        ]
+
+        runner = CustomRunner()
+
+        # model training
+        runner.train(
+            model=model,
+            optimizer=optimizer,
+            loaders=loaders,
+            logdir="./logs",
+            num_epochs=5,
+            verbose=True,
+            callbacks=[BatchTransformCallback(transrorms, input_key=0)],
+        )
+
+    To apply augmentations only during specific loader e.g. only during
+    training :class:`catalyst.core.callbacks.control_flow.ControlFlowCallback`
+    callback can be used. For config API it can look like this:
+
+    .. code-block:: yaml
+
+        callbacks_params:
+          ...
+          train_transforms:
+            _wrapper:
+              callback: ControlFlowCallback
+              loaders: train
+            callback: BatchTransformCallback
+            transforms:
+              - transform: kornia.RandomAffine
+                degrees: [-15, 20]
+                scale: [0.75, 1.25]
+                return_transform: true
+              - transform: kornia.ColorJitter
+                brightness: 0.1
+                contrast: 0.1
+                saturation: 0.1
+                return_transform: false
+            input_key: image
+            additional_input_key: mask
+          ...
+
+    .. _`Kornia: an Open Source Differentiable Computer Vision Library
+        for PyTorch`: https://arxiv.org/pdf/1910.02190.pdf
+    """
+
+    def __init__(
+        self,
+        transform: Sequence[Union[dict, AugmentationBase]],
+        input_key: Union[str, int] = "image",
+        additional_input_key: Optional[str] = None,
+        output_key: Optional[Union[str, int]] = None,
+        additional_output_key: Optional[str] = None,
```

Comment:
could we support `input_key : Union[str, Tuple[str]] ` to remove `additional_input_key` dependency?

Reply:
Should we?
Kornia transforms can be applied only to only one and two targets, so additional checks dependent on the type on the input keys will be needed.